aea/protocols/dialogue/base.py

```diff
@@ -720,7 +726,7 @@ def reply(
 
         reply = self._message_class(
             dialogue_reference=self.dialogue_label.dialogue_reference,
-            message_id=last_message.message_id + 1,
+            message_id=self.get_next_message_id(),
```

Comment:
A reply needs to use a `message_id` unused by and unknown  to `self`. So the `message_id` needs to be the max of all messages ids (incoming and outgoing) + 1.